river/anomaly/xstream.py

```diff
@@ -0,0 +1,220 @@
+import anomaly
+        
+from utils.math import get_minmax_array
+#import math     #import math pour utiliser get_minmax_array
+
+#from utils import get_minmax_array
+from river.utils import dict2numpy
+import numpy as np
+
+
+from transform.projection.streamhash_projector import StreamhashProjector
+#from pysad.utils import get_minmax_array
+
+
+class xStream(anomaly.base.AnomalyDetector):
+    """The xStream model for row-streaming data :cite:`xstream`. It first projects the data via streamhash projection. It then fits half space chains by reference windowing. It scores the instances using the window fitted to the reference window.
+    Args:
+        n_components (int): The number of components for streamhash projection (Default=100).
+        n_chains (int): The number of half-space chains (Default=100).
+        depth (int): The maximum depth for the chains (Default=25).
+        window_size (int): The size (and the sliding length) of the reference window (Default=25).
```

Comment:
As you can see in the rest of River's codebase, we don't use this format for documenting class parameters.